editor/src/components/canvas/canvas-strategies/strategies/resize-grid-strategy.ts

```diff
@@ -149,7 +172,14 @@ export const resizeGridStrategy: CanvasStrategyFactory = (
           ),
         mergedValues.dimensions,
       )
-      const propertyValueAsString = printArrayGridDimension(newSetting)
+
+      const newDimensions = buildResizedDimensionsArray({
```

Comment:
I haven't read the code in detail yet but _it feels like_ what is happening is that `resizedDimensions` is trash and you wrote an elaborate helper function `buildResizedDimensionsArray` to essentially untrash it. can't we make it so that `resizedDimensions` is not weird to begin with? I feel like `buildResizedDimensionsArray` could be remade to do the actual resizing if you would give it a single `resizedValue` instead of all of `resizedDimensions`, and then we could ditch `getNewDragValue`

Reply:
Makes sense, updated the code to replace directly instead 👍